src/operations.jl

```diff
@@ -5,25 +5,9 @@ hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
 # rand
-typealias RandTypes{T,C} Union{AbstractRGB{T},
-                               TransparentRGB{C,T},
-                               AbstractGray{T},
-                               TransparentGray{C,T}}
-typealias RandTypesFloat{C,T<:AbstractFloat} RandTypes{T,C}
-
-rand{G<:AbstractGray}(::Type{G}) = G(rand())
-rand{G<:TransparentGray}(::Type{G}) = G(rand(), rand())
-rand{C<:AbstractRGB}(::Type{C}) = C(rand(), rand(), rand())
-rand{C<:TransparentRGB}(::Type{C}) = C(rand(), rand(), rand(), rand())
-function rand{C<:RandTypesFloat}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(eltype(C), (sizeof(C)÷sizeof(eltype(C)), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f8}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt8, (sizeof(C), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f16}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt16, (sizeof(C)>>1, sz...)), sz)
-end
+rand{T<:Colorant}(::Type{T}) = reinterpret(T, rand(eltype(T), div(sizeof(T), sizeof(eltype(T)))))[1]
```

Comment:
Hmm, the performance implications are somewhat concerning. Maybe make the RHS
```julia
mapc(x->rand(eltype(T)), base_colorant_type(T)())
```
If we let `myrand` be yours and `myrand3` be mine, I get
```julia
julia> @benchmark myrand(HSV{Float32})
BenchmarkTools.Trial: 
  samples:          10000
  evals/sample:     759
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  352.00 bytes
  allocs estimate:  6
  minimum time:     155.00 ns (0.00% GC)
  median time:      164.00 ns (0.00% GC)
  mean time:        213.86 ns (21.27% GC)
  maximum time:     5.59 μs (93.40% GC)

julia> @benchmark myrand3(HSV{Float32})
BenchmarkTools.Trial: 
  samples:          10000
  evals/sample:     997
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  0.00 bytes
  allocs estimate:  0
  minimum time:     21.00 ns (0.00% GC)
  median time:      22.00 ns (0.00% GC)
  mean time:        22.39 ns (0.00% GC)
  maximum time:     70.00 ns (0.00% GC)
```